selfapi.js

```diff
@@ -367,6 +374,24 @@ API.prototype = {
     });
   },
 
+  // Build index routes.
+  buildIndexRoutes: function (basePath) {
+      var fullPath = normalizePath(this.path, basePath) || '/';
+      this.get({
+          title: 'Index',
```

Comment:
Nit: I don't really like the fact that this function implements an opinionated API handler directly on itself, without giving you control over handler name, description, path, method, examples or response format. Maybe it would be more useful to make this a simple exporter, e.g. something like `toAPIIndex()` or `toChildrenIndex` or `toIndexRoutes` etc? Then API implementors could simply do something like:

```js
api.get({
  name: `Get API Index`,
  description: `Helpful in some cases`,
  handler: (req, res) => {
    res.json(api.toAPIIndex(), null, 2);
  },
  examples: []
}
```